### DISCUSSION TITLE
Question about the security of NPM packages published in the Verdaccio private registry

### DISCUSSION THREAD

**Discussion by WangJiaoJo** [verdaccio/verdaccio#3721] [Q&A]
Hi, hope you are doing well!

I have a question about the NPM packages security in the private registry built by Verdaccio.
For example, if I build a private registry using Verdaccio on server side and publish an NPM package to this private registry, will the NPM package only be cached on this server? Will it be pushed or cached on any cloud side Verdaccio server side?

I'm looking forward to your reply. Thanks.

**Comment by juanpicado** [comment]
> For example, if I build a private registry using Verdaccio on server side and publish an NPM package to this private registry, will the NPM package only be cached on this server? 

yes, if you follow the best practice guidelines.

> Will it be pushed or cached on any cloud side Verdaccio server side?

Not sure if I follow, but I'll assume you mean that if you publish to your private instance the package will be routed or re-published somewhere else, the answer is no. And could you elaborate what means ` cloud side Verdaccio server side?`

**Reply by WangJiaoJo** [reply]
Hi, Juanpicado,

Thanks for your reply! I was wondering if we publish the NPM packages to Verdaccio private registry, will the packages be stored to Verdaccio server? Thanks!

**Reply by juanpicado** [reply]
Yes, that's the main point to use verdaccio as private registry. What other behaviours you were thinking might happen? 🤔

**Reply by WangJiaoJo** [reply]
Hi, Juanpicado,
I am clear now. Thanks so much for your anwser!
